### ISSUE TITLE
Baseball out-of-sample example is misleading

### ISSUE THREAD

**Issue by kiwidamien** [pyro-ppl/numpyro#993]
# What

Example showing prediction on out-of-sample assumes that the size of the test set is the same as size of the training set, not typically true in ML models.

## Detail

The [baseball](http://num.pyro.ai/en/stable/examples/baseball.html) example shows how to train several predictive models using various degrees of pooling on a training set, and apply to a new set of (out-of-sample) baseball players. If you add or remove players from the testing set, you will get broadcasting errors.

## Recommendation

* Docs have lots of examples that are inference/shrinkage applied to in-sample documentation (e.g. Waffle house)
* The PDF talks about using "guides" in the `Predictive` function to apply to out-of-sample, but it is hard to find simple examples. It would be nice (but a lot of work) to use this example to show how to use guides.

If not using guides, I would recommend stating prominiently that this is a trick that assumes the training and testing sets have the same number of rows.

# Where
Looking at example here:
* [Python file](https://github.com/pyro-ppl/numpyro/blob/master/examples/baseball.py)
* [Documentation](http://num.pyro.ai/en/stable/examples/baseball.html)

**Comment by fehiepsi** [comment]
Yes, the example only shows how to give predictions for the same players over the remaining season. The name "no pooling" means that you are inferring each set of parameters for each player. If you want to use "no pooling" model for new players, you just run MCMC to infer parameters for those new players, then make predictions.

We are still lacking tutorials/examples of using [SVI in numpyro](http://num.pyro.ai/en/stable/svi.html#numpyro.infer.svi.SVI) (see how `guide` is used there and in [Pyro SVI tutorials](http://pyro.ai/examples/svi_part_i.html)). For now, you can see how to use `guide` in `Predictive` in [this Pyro example](http://pyro.ai/examples/bayesian_regression.html#Model-Evaluation).

**Comment by kiwidamien** [comment]
Ah awesome, thanks for the pointer!

(And overall this project is awesome!)

**Comment by fehiepsi** [comment]
Thanks, glad that you find the library helpful! :)

**Comment by fehiepsi** [comment]
I guess the current description in the example is clear enough but please feel free to reopen this to make the description clearer.
